src/main/java/io/confluent/copycat/hdfs/partitioner/FieldPartitioner.java

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright 2015 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ **/
+
+package io.confluent.copycat.hdfs.partitioner;
+
+import org.apache.kafka.copycat.data.Schema;
+import org.apache.kafka.copycat.data.Schema.Type;
+import org.apache.kafka.copycat.data.Struct;
+import org.apache.kafka.copycat.errors.CopycatException;
+import org.apache.kafka.copycat.sink.SinkRecord;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+
+import io.confluent.copycat.hdfs.HdfsSinkConnectorConfig;
+
+public class FieldPartitioner implements Partitioner {
+  private static final Logger log = LoggerFactory.getLogger(FieldPartitioner.class);
+  private String fieldName;
+
+  @Override
+  public void configure(Map<String, Object> config) {
+    fieldName = (String) config.get(HdfsSinkConnectorConfig.PARTITION_FIELD_CONFIG);
+  }
+
+  @Override
+  public String encodePartition(SinkRecord sinkRecord) {
+    Object value = sinkRecord.value();
+    Schema valueSchema = sinkRecord.valueSchema();
+    if (value instanceof Struct) {
+      Struct struct = (Struct) value;
+      Object partitionKey = struct.get(fieldName);
+      Type type = valueSchema.field(fieldName).schema().type();
+      switch (type) {
+        case INT8:
+        case INT16:
+        case INT32:
+        case INT64:
```

Comment:
float32, float64? Not sure how complete these needs to be. Long term, it might make sense to allow for settings like `PARTITION_FIELD_CONFIG = "user.name"` which is automatically turned into field/map entry lookups. Certainly ok to skip for now, but might be worth looking into what other systems support wrt partitioning to see whether something like that would be useful.